File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    minPrice := prices[0]
    ans := 0
    for _,value := range prices {
        if(value - minPrice > ans){
            ans = value - minPrice
        }
        if(value < minPrice){
            minPrice = value
        }
    }
    return ans
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int minm = prices[0],n = prices.size();
        for(int i=1;i<n;i++){
            ans = max(ans,prices[i]-minm);
            minm = min(minm,prices[i]);
        }
        return ans;
    }
};